app/models/beta/search/search_result.rb

```diff
@@ -0,0 +1,64 @@
+module Beta
+  module Search
+    class SearchResult
+      delegate :id, to: :search_query_parser_result, prefix: true
+
+      attr_accessor :took,
+                    :timed_out,
+                    :hits,
+                    :max_score,
+                    :search_query_parser_result
+
+      class << self
+        def build(result, search_query_parser_result)
+          search_result = new
+
+          search_result.took = result.took
+          search_result.timed_out = result.timed_out
+          search_result.max_score = result.hits.max_score
+          search_result.hits = result.hits.hits.map(&method(:build_hit))
+          search_result.search_query_parser_result = search_query_parser_result
+
+          search_result
+        end
+
+        def build_hit(hit_result)
```

Comment:
Maybe if we change the name of the module to OpenSearch, then the term HIT acquires meaning, otherwise (without context) it looks a bit obscure.

Reply:
I've opted to name the SearchResult OpenSearchResult rather than changing the namespace